monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -1230,9 +1230,11 @@ sealed abstract class Task[+A] extends Serializable {
     *
     * The application of this function has strict behavior, as the
     * task is immediately executed.
+    *
+    * Exceptions in `f` are reported using provided (implicit) Scheduler
     */
-  final def foreach(f: A => Unit)(implicit s: Scheduler): CancelableFuture[Unit] =
-    foreachL(f).runAsync(s)
+  final def foreach(f: A => Unit)(implicit s: Scheduler): Unit =
+    runAsync.foreach(f)
```

Comment:
We should also add a `@deprecated` tag.

Reply:
@alexandru I actually wanted your opinion on this. While I myself don't need `foreach` at all, this should fix all problems we discussed at #619, so there might not be a big problem if it was to stay.

Reply:
I don't know, maybe we should leave it.

I also don't like that we are changing the signature, because it creates yet another incompatibility. An alternative implementation would be something like ...

```scala
task.transform(_ => (), s.reportFailure).runAsync
```

And it keeps the `CancelableFuture[Unit]`. We don't necessarily have to `@deprecate` it now.

Reply:
@alexandru but that doesn't fix the for-comprehension, does it?

Reply:
What do you mean? 

The problem was that it doesn't report errors via `Scheduler#reportFailure` like `Future` does, a behavior that apparently people expect.

Have you noticed other problems?

Reply:
@alexandru 
> ... the result will not wait on an inner-for, this isn't a flatMap that we are describing, so returning a Future in here gives a false sense of what the thing does.
https://github.com/monix/monix/issues/619#issuecomment-375573445

Reply:
^ I thought from that perspective it makes sense to return `Unit`.

Reply:
Ah, right. I guess we can leave it with `Unit`.